Desktop/widgets/listmodelfiltereddataentry.cpp

```diff
@@ -331,36 +332,37 @@ QVariant ListModelFilteredDataEntry::data(const QModelIndex &index, int role) co
 	if(column == _editableColumn)
 		return QVariant(_values[0][row]);
 
-	if(!DataSetPackage::pkg()->hasDataSet() || column > _colNames.size() || column < 0)
+	if(getDataSetRowCount() == 0 || column > _colNames.size() || column < 0)
 		return QVariant();
 
 	std::string colName = _colNames[column].toStdString();
 	size_t rowData		= _filteredRowToData[static_cast<size_t>(row)];
 
-	DataSetPackage * package = DataSetPackage::pkg();
+	ColumnsModel* columnsModel = AnalysisForm::getColumnsModel();
```

Comment:
See, if you had something like `ColumnsModel::colModel()` you could just use that directly!

Reply:
And if it was specific to each AnalysisForm it ought to not be static.